client/src/socket.js

```diff
@@ -3,130 +3,154 @@ import Victor from 'victor';
-    if (sg.pickups[pickupData.id]) {
-      sg.pickups[pickupData.id].setData(pickupData);
-    } else {
-      sp.gameObjects[pickupData.id] = sg.pickups[pickupData.id] = new Pickup(s)
-        .addCollider(new CircleCollider())
-        .addRenderer(new SpriteRenderer())
-        .setData({
-          ...pickupData,
-          renderer: { radius: 1, spriteName: pickupData.pickupType }
-        });
-    }
-  }
-  
+
   socket.on('allPickups', (pickupsData) => {
-    Object.keys(sg.pickups).forEach(pickupId => {
-      delete sp.gameObjects[pickupId];
-      delete sg.pickups[pickupId];
-    });
-    pickupsData.forEach(pickupData => {
-      updatePickup(pickupData);
+    sg.pickups = {};
+    pickupsData.forEach(data => {
+      sg.pickups[data.id] = new Pickup(data.pickupType, data.x, data.y);
     });
   });
-  
-  socket.on('updatePickup', (pickupData) => {
-    updatePickup(pickupData);
+
+  socket.on('updatePickup', (data) => {
+    let pickup = sg.pickups[data.id];
+    if (pickup) {
+      pickup.pickupType = data.pickupType;
+      pickup.x = data.x;
+      pickup.y = data.y;
+    } else {
+      sg.pickups[data.id] = new Pickup(data.pickupType, data.x, data.y);
+    }
   });
-  
-  socket.on('collectedPickup', ({ clientId, pickupId, worth }) => {
-    if (!sg.pickups[pickupId]) return;
-    if (clientId && clientId != sg.clientId) {
-      sg.players[clientId].score += sg.pickups[pickupId].worth;
-      sg.players[clientId].collider.updateBodyWithScore()
+
+  socket.on('collectedPickup', ({ playerId, pickupId, score }) => {
+    if (!sg.pickups[pickupId]) {
+      console.debug('pickup collected that wasn\'t synced', playerId, pickupId);
+    }
+    if (playerId && playerId != sg.clientId) {
+      //we assume client players stay in sync without these updates
+      let pl = sg.players[playerId];
+      pl.score = score;
+      pl.updateBodyWithScore();
+      sortScoreboard(sg.scoreboard);
     }
-    delete sp.gameObjects[pickupId];
     delete sg.pickups[pickupId];
   });
 
   //Listen for game state changes
-  socket.on("updateGameState", (newState) => {
+  socket.on('updateGameState', (newState) => {//TODO: send these updates with data depending on the current game state
     //Update things based on server-authoritative state
-    s.game.gameState = newState.gameState;
-    s.game.gameStateTimer = newState.gameStateTimer;
-    s.game.scoreboard = newState.scoreboard?.sort(([_, __, plScore1], [___, ____, plScore2]) => plScore1 > plScore2 ? -1 : 1);
-    s.score.winner = newState.winnerID;
-    
+    sg.gameState = newState.gameState;
+    sg.gameStartTimer = newState.gameStartTimer;
+    sg.gameEndTimer = newState.gameEndTimer;
+    sg.roundTimer = newState.roundTimer;
+    sg.serverTime = newState.serverTime;
+    // s.score.winner = newState.winnerID;
+
     //Update players
     //Loop through every player in the data
-    for (const clientId in newState.players) {
+    for (const data in newState.players) {
```

Comment:
Is players an array now? I'm surprised this works. for ... in loops in js assign the key to the variable, not the value